### ISSUE TITLE
`@module` causes false `call-non-callable`

### ISSUE THREAD

**Issue by lewis6991** [EmmyLuaLs/emmylua-analyzer-rust#1238]
Version: `emmylua_check 0.25.1`, default configuration.

`callable.lua`:

```lua
---@param value string
---@return string
local function callable(value)
  return value
end

return callable
```

`main.lua`:

```lua
---@module 'callable'
local callable = require('callable')
assert(callable('hello') == 'hello')
```

Running `emmylua_check . --warnings-as-errors` reports:

```text
main.lua:3:8
Cannot call expression of type `fun(value: string) -> string`. [call-non-callable]
```

Expected: no diagnostic. Removing the `@module` annotation makes the check pass.
